src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -354,21 +302,19 @@ export class FocusMonitor implements OnDestroy {
     }
   }
 
-  private _getFocusOrigin(event: FocusEvent): FocusOrigin {
-    // If we couldn't detect a cause for the focus event, it's due to one of three reasons:
-    // 1) The window has just regained focus, in which case we want to restore the focused state of
-    //    the element from before the window blurred.
-    // 2) It was caused by a touch event, in which case we mark the origin as 'touch'.
-    // 3) The element was programmatically focused, in which case we should mark the origin as
-    //    'program'.
+  private _getFocusOrigin(): FocusOrigin {
     if (this._origin) {
       return this._origin;
     }
 
+    // If we couldn't determine a focus origin, it's because either:
+    // 1) The window has just regained focus, and thus we restore the last origin from before the
+    //    window blurred.
+    // 2) The element was programmatically focused, and thus we set the origin as 'program'.
+    // 3) The element was focused via navigating with a screen reader, and thus we set the origin
+    //    as 'program' because keyboard events are rarely fired.
```

Comment:
I don't totally follow the part here about keyboard events being rarely fired. Wouldn't screen reader navigation almost always be keyboard navigation?

Reply:
When navigating with JAWS/NVDA/VO, no keyboard events are fired (for the most part). For example, if a user is linearly navigating with VO via Caps Lock + Arrow Keys, no keyboard events are fired _despite focus moving along with VO's virtual cursor_.  As another example, if a user is focused on element A, then uses JAWS' browse mode to navigate to element B (note focus doesn't yet move in this case), and then presses Enter to enter forms mode and move focus to B, again, no keyboard events are fired _despite focus moving_. 

In these scenarios, we can't detect the input modality as keyboard, and as a result, any subsequent focus change will be attributed to program. Note that this is `FocusMonitor`'s existing behavior today, it just isn't documented anywhere.

Reply:
Got it- I knew screen reader interaction didn't kick off keyboard events, I just misunderstood the comment (the point being that you can't distinguish between programatic and screen-reader). I might reword slightly:
```ts
// If the window has just regained focus, we can restore the most recent origin
// from before the window blurred. Otherwise, we've reached the point where
// we can't identify the source of the focus. This typically means one of two things
// happened:
// 1) The element was programmatically focused, or
// 2) The element was focused via screen reader navigation (which
// generally doesn't fire events).
// Because we can't distinguish between these two cases, we default to
// setting `program`, erring on the side of not showing keyboard focus indicators.
```

Reply:
Done (but omitted the final piece "erring on the side of not showing keyboard focus indicators" because FocusMonitor doesn't know about keyboard focus indicators, and can be used with or without them).